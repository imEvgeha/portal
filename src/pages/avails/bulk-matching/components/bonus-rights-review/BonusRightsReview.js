import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {Button} from '@portal/portal-components';
import {get} from 'lodash';
import TitleMatchingRightsTable from '../../../title-matching-rights-table/TitleMatchingRightsTable';
import './BonusRightsReview.scss';

const tabs = {
    bonusRights: 'Created Bonus Rights',
    existingBonusRights: 'Existing Bonus Rights',
};

const BonusRightsReview = props => {
    const {closeDrawer} = props;
    const [activeTab, setActiveTab] = useState('bonusRights');
    // eslint-disable-next-line react/prop-types
    const Tab = ({tab}) => (
        <div
            className={`nexus-c-bonus-rights-review__tabs--tab
        ${activeTab === tab ? 'active-tab' : ''}`}
            onClick={() => setActiveTab(tab)}
        >
            {`${tabs[tab]} (${get(props, `${tab}.length`, 0)})`}
        </div>
    );
    return (
        <div className="nexus-c-bonus-rights-review">
            <div className="nexus-c-bonus-rights-review__tabs">
                {Object.keys(tabs).map(tab => (
                    <Tab tab={tab} key={tab} />
                ))}
            </div>
            <TitleMatchingRightsTable data={get(props, activeTab, [])} />
            {closeDrawer && (
                <Button label="Done" onClick={closeDrawer} className="p-button-outlined p-button-secondary" />
            )}
        </div>
    );
};

BonusRightsReview.propTypes = {
    closeDrawer: PropTypes.func,
};

BonusRightsReview.defaultProps = {
    closeDrawer: null,
};

export default BonusRightsReview;
