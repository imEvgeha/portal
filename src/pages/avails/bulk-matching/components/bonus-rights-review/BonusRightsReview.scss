@import '~@vubiquity-nexus/portal-styles/scss/config.scss';

.nexus-c-bonus-rights-review {
    display: flex;
    flex-direction: column;
    margin-top: 20px;
    button {
        align-self: flex-end;
        margin-top: 20px;
    }
    &__tabs {
        display: flex;
        &--tab {
            margin-right: 20px;
            cursor: pointer;
            font-size: 16px;
            color: map-get($colors, nexus-light-grey);

            &.active-tab {
                pointer-events: none;
                color: map-get($colors, nexus-title-color);
            }
        }
    }
}
