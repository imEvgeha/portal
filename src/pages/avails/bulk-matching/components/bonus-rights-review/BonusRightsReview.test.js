import React from 'react';
import {shallow} from 'enzyme';
import BonusRightsReview from './BonusRightsReview';

describe('BonusRightsReview', () => {
    let wrapper = null;
    const props = {
        bonusRights: [],
        existingBonusRights: [],
        closeDrawer: null,
    };
    beforeAll(() => {
        wrapper = shallow(<BonusRightsReview {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render created tab at first', () => {
        const firstTab = wrapper.find('Tab').at(0).shallow().find('.nexus-c-bonus-rights-review__tabs--tab');
        expect(firstTab.hasClass('active-tab')).toBe(true);
    });
});
