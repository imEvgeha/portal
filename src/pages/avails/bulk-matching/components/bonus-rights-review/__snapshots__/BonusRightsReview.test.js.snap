// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BonusRightsReview should match snapshot 1`] = `
<div
  className="nexus-c-bonus-rights-review"
>
  <div
    className="nexus-c-bonus-rights-review__tabs"
  >
    <Tab
      key="bonusRights"
      tab="bonusRights"
    />
    <Tab
      key="existingBonusRights"
      tab="existingBonusRights"
    />
  </div>
  <TitleMatchingRightsTable
    data={Array []}
  />
</div>
`;
